### PR TITLE
Correct project_root path for ProjectJson.

### PR DESCRIPTION
**Pull request by woody77**
It was already the folder containing the rust-project.json file, not the file itself.  This also removes the Option-ness of it, since it's now an infallible operation to set the member value.

### CODE REVIEW

File: crates/project_model/src/project_json.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ impl ProjectJson {
     pub fn new(base: &AbsPath, data: ProjectJsonData) -> ProjectJson {
```
Comment (jonas-schievink) [inline crates/project_model/src/project_json.rs:37]: Can you add a doc comment here that explains where `base` should point to when calling it?

File: crates/project_model/src/project_json.rs [hunk 2]
```diff
@@ -91,11 +91,8 @@ impl ProjectJson {
     pub fn crates(&self) -> impl Iterator<Item = (CrateId, &Crate)> + '_ {
         self.crates.iter().enumerate().map(|(idx, krate)| (CrateId(idx as u32), krate))
     }
-    pub fn path(&self) -> Option<&AbsPath> {
-        match &self.project_root {
-            Some(p) => Some(p.as_path()),
-            None => None,
-        }
+    pub fn path(&self) -> &AbsPath {
```
Comment (jonas-schievink) [inline crates/project_model/src/project_json.rs:94]: (short doc comment on this method would be nice too)

File: crates/project_model/src/project_json.rs (hunk 1 above)
Comment (woody77) [reply crates/project_model/src/project_json.rs:37]: Certainly.  I found this by seeing what the resultant `pwd` to a started command was, and realizing that the path here is to the workspace folder that contains rust-project.json, not the path to rust-project.json itself.

Comment (woody77) [reply crates/project_model/src/project_json.rs:44]: Done.

File: crates/project_model/src/project_json.rs (hunk 2 above)
Comment (woody77) [reply crates/project_model/src/project_json.rs:104]: Done.

**Comment by jonas-schievink** [comment]
Thanks!

bors r+

**Pull request merged**